app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseDatabase.kt

```diff
@@ -297,6 +300,94 @@ class FirebaseDatabase(activity: Activity) : Database {
     override suspend fun unfollow(follower: String, followee: String) {
         doFollow(follower, followee, follow = false)
     }
-}
 
+    /**
+     * Returns a list of users that liked a given challenge
+     * @param cid the challenge id
+     * @return a list of users that liked the challenge
+     */
+    internal fun getLikesOf(cid: String): Task<Map<String, Boolean>> {
+        return dbChallengeRef.child(cid).child("likes").get()
```

Comment:
Pay attention `dbChallengeRef.child(cid)` does not work because cid has actually two components internally. Please check how this is done within `FirebaseChallengeRef`. Otherwise good